src/database/mixxxdb.cpp

```diff
@@ -0,0 +1,86 @@
+#include "database/mixxxdb.h"
+
+#include "database/schemamanager.h"
+
+#include "util/assert.h"
+#include "util/logger.h"
+
+
+// The schema XML is baked into the binary via Qt resources.
+//static
+const QString MixxxDb::kDefaultSchemaFile(":/schema.xml");
+
+//static
+const int MixxxDb::kRequiredSchemaVersion = 27;
+
+namespace {
+
+const mixxx::Logger kLogger("MixxxDb");
+
+// The connection parameters for the main Mixxx DB
+mixxx::DbConnection::Params dbConnectionParams(
+        const UserSettingsPointer& pConfig) {
+    mixxx::DbConnection::Params params;
+    params.type = "QSQLITE";
+    params.hostName = "localhost";
+    params.filePath = QDir(pConfig->getSettingsPath()).filePath("mixxxdb.sqlite");
+    params.userName = "mixxx";
+    params.password = "mixxx";
+    return params;
+}
+
+} // anonymous namespace
+
+MixxxDb::MixxxDb(
+        const UserSettingsPointer& pConfig)
+    : m_pDbConnectionPool(std::make_shared<mixxx::DbConnectionPool>(dbConnectionParams(pConfig), "MIXXX")) {
```

Comment:
I did expected that we use create() here. On a second thought calling the constructor seams to be somehow more reasonable if we get rid of the static data in DbConnectionPool